sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -55,7 +56,7 @@ object SQLConf {
   val WAREHOUSE_PATH = SQLConfigBuilder("spark.sql.warehouse.dir")
     .doc("The default location for managed databases and tables.")
     .stringConf
-    .createWithDefault("${system:user.dir}/spark-warehouse")
+    .createWithDefault("${system:user.dir}" + File.separator + "spark-warehouse")
```

Comment:
Although the path points to a local dir, this path is actually a hadoop filesystem path. So, I am wondering if we need to use `File.separator` at here? If we are running on Windows, does a hadoop filesystem path uses `\` as the separator?